api/ccapi/include/model.h

```diff
@@ -95,14 +125,36 @@ class Model {
 
   /**
    * @brief     save model and training parameters into file
+   * @todo      deprecate this
```

Comment:
how about [[deprecate]]??

Reply:
We are not allowing having deprecated warning, until we decide to change let's do in this fashion :p

Reply:
aah, got it. Then before M2 release, we will remove the deprecated APIs right?

Reply:
Yes